include/dsn/dist/replication/replication_client_base.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Microsoft Corporation
+ *
+ * -=- Robust Distributed System Nucleus (rDSN) -=-
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/*
+ * Description:
+ *     interface for clients operations using rDSN
+ *
+ * Revision history:
+ *     Dec., 2015, @xiaotz (Xiaotong Zhang), first version
+ *     xxxx-xx-xx, author, fix bug about xxx
+ */
+
+
+#pragma once
+
+//
+// replication_client_base is the base class for clients to access
```

Comment:
this is not consistent with our code generation process, with which we get a app client with very simple interface (regarding to .thrift file). Please take a look at what we have in counter service and in rocksdb. In that case, this class is impossible as it creates replication-app-base automatically without knowing their types. In fact, I don't see the necessity of having such as manager class - it brings more complexities and deeper stack with runtime cost. Please elaborate the necessity if i'm missing sth.